// @6d-ui/popup/example/src/components/Create.js
import React, { Component } from "react";
import {
    ModalBody,
    ModalFooter
} from 'reactstrap';
import {
  CustomButton,
  BUTTON_STYLE,
  BUTTON_TYPE,
  BUTTON_SIZE,
  BUTTON_ALIGN
} from '6d-buttons'

class Create extends Component {
  constructor(props) {
    super(props);
    this.state = {
      windowHeight : window.innerHeight
    }
    this.updateWindowDimensions = this.updateWindowDimensions.bind(this);
  }
  componentDidMount() {
    this.updateWindowDimensions();
    window.addEventListener("resize", this.updateWindowDimensions);
  }
  componentWillUnmount() {
    window.removeEventListener("resize", this.updateWindowDimensions);
  }
  updateWindowDimensions() {
    this.setState({ windowHeight: window.innerHeight });
  }

  render() {
    const height = {
      height: this.state.windowHeight - 135
    };
    return (
      <ModalBody>
        <div className="overlay_position scrollbar" style={height}>
          <div style={{height : '1000px'}}>
            SDadsd
          </div>
          End
        </div>
        <ModalFooter>
          <CustomButton
            style={BUTTON_STYLE.BRICK}
            type={BUTTON_TYPE.SECONDARY}
            size={BUTTON_SIZE.LARGE}
            align="right"
            label="Reset"
            isButtonGroup={true}
          />
          <CustomButton
            style={BUTTON_STYLE.BRICK}
            type={BUTTON_TYPE.PRIMARY}
            size={BUTTON_SIZE.LARGE}
            align="right"
            label="Create"
            isButtonGroup={true}
          />
        </ModalFooter>
      </ModalBody>
    );
  }
}

export default Create;
