// @6d-ui/ui-components/build/index.js
'use strict';

var _SideMenu = require('./components/sideMenu/SideMenu');

var _SideMenu2 = _interopRequireDefault(_SideMenu);

var _Header = require('./components/header/Header');

var _Header2 = _interopRequireDefault(_Header);

var _Notification = require('./components/notification/Notification');

var _Notification2 = _interopRequireDefault(_Notification);

var _Alert = require('./elements/alert/Alert');

var _Alert2 = _interopRequireDefault(_Alert);

var _Loader = require('./elements/loader/Loader');

var _Loader2 = _interopRequireDefault(_Loader);

var _ActionLoader = require('./elements/actionLoader/ActionLoader');

var _ActionLoader2 = _interopRequireDefault(_ActionLoader);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

module.exports = {
  SideMenu: _SideMenu2.default,
  Header: _Header2.default,
  Notification: _Notification2.default,
  Alert: _Alert2.default,
  Loader: _Loader2.default,
  ActionLoader: _ActionLoader2.default
};
