// @6d-ui/popup/build/elements/Popup.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Types = require('../constants/Types');

var _reactstrap = require('reactstrap');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Popup = function Popup(props) {
  var className = 'custom-modal ' + (props.type && props.type === _Types.POPUP_ALIGN.RIGHT ? 'right-fixed-modal' : 'center-modal') + (props.className ? ' ' + props.className : '');
  var getMinWidth = function getMinWidth() {
    if (props.minWidth) {
      return {
        minWidth: props.minWidth
      };
    }
  };
  var getWidth = function getWidth() {
    if (props.width) {
      return {
        width: props.width
      };
    }
  };
  var getHeader = function getHeader() {
    if (props.isCustomTitle) {
      return _react2.default.createElement(
        'div',
        null,
        props.title
      );
    } else {
      return _react2.default.createElement(
        _reactstrap.ModalHeader,
        { className: 'primary-background', toggle: function toggle() {
            return props.close(0);
          } },
        props.title
      );
    }
  };
  return _react2.default.createElement(
    _reactstrap.Modal,
    { style: _extends({}, getMinWidth(), getWidth()), className: className, isOpen: props.isOpen, toggle: function toggle() {
        return props.close(0);
      } },
    getHeader(),
    props.component
  );
};
exports.default = Popup;
