// @6d-ui/ui-components/build/elements/alert/Alert.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _popup = require('@6d-ui/popup');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Alert = function Alert(_ref) {
  var setModalPopup = _ref.setModalPopup,
      _ref$modal = _ref.modal,
      modal = _ref$modal === undefined ? { isOpen: false } : _ref$modal;
  var isOpen = modal.isOpen,
      title = modal.title,
      content = modal.content,
      CancelBtnLabel = modal.CancelBtnLabel,
      confirmBtnLabel = modal.confirmBtnLabel,
      rowId = modal.rowId,
      onConfirmCallBack = modal.onConfirmCallBack;

  var toggle = function toggle() {
    setModalPopup({ isOpen: false });
  };
  return _react2.default.createElement(_popup.BasicAlert, {
    isOpen: isOpen,
    title: title,
    content: content,
    CancelBtnLabel: CancelBtnLabel,
    confirmBtnLabel: confirmBtnLabel,
    rowId: rowId,
    close: toggle,
    onConfirmCallBack: onConfirmCallBack
  });
};
exports.default = Alert;
