// @6d-ui/popup/example/src/App.js
import React, { Component } from 'react';
import './App.css';
import Create from './components/Create';
import Search from './components/Search';
import {Popup, POPUP_ALIGN, BasicAlert} from '6d-action-popup';
import {
    ModalBody,
    ModalFooter
} from 'reactstrap';

class App extends Component {
  constructor(props) {
    super(props);
    this.toggleAction = this.toggleAction.bind(this);
    this.state = {
      modal: 0
    }
  }
  toggleAction(type) {
    this.setState({modal : type})
  }
  onConfirmCallBack(rowId) {
    console.log(rowId);
  }
  render() {
    return (
      <div className="App">
        <div onClick={()=>this.setState({modal : 2})}>Open Right Modal</div>
        <div onClick={()=>this.setState({modal : 1})}>Open Normal Modal</div>
        <div onClick={()=>this.setState({modal : 3})}>Open Alert Box Modal</div>
        <Popup
          type={POPUP_ALIGN.RIGHT}
          title="Create"
          isOpen={this.state.modal === 2}
          close={this.toggleAction}
          component={
            <Create
            />
          }
        />
        <Popup
          type={POPUP_ALIGN.CENTER}
          title="Search"
          isOpen={this.state.modal === 1}
          close={this.toggleAction}
          component={
            <Search
            />
          }
        />
        <BasicAlert
          isOpen={this.state.modal === 3}
          title="Confirm Delete"
          content="Are You Sure Delete Id 1 ?"
          CancelBtnLabel="Cancel"
          confirmBtnLabel="Confirm"
          rowId="1"
          close={this.toggleAction}
          onConfirmCallBack={this.onConfirmCallBack.bind(this)}
        />

      </div>
    );
  }
}

export default App;
