// @6d-ui/ui-helpers/build/components/routeVisits/FrequencyTab.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactstrap = require('reactstrap');

var _lodash = require('lodash');

var _lodash2 = _interopRequireDefault(_lodash);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var FrequencyTab = function (_Component) {
    _inherits(FrequencyTab, _Component);

    function FrequencyTab() {
        _classCallCheck(this, FrequencyTab);

        return _possibleConstructorReturn(this, (FrequencyTab.__proto__ || Object.getPrototypeOf(FrequencyTab)).apply(this, arguments));
    }

    _createClass(FrequencyTab, [{
        key: 'render',
        value: function render() {
            var _this2 = this;

            var getFrequency = function getFrequency(list, type) {
                return list.map(function (cal, index) {
                    var freqList = type === 1 ? _this2.props.weekDaysList : _this2.props.weekList;
                    var freqClass = 'route-frequency ' + (_lodash2.default.includes(freqList, cal.value) ? 'route-frequency-selected' : "");
                    return _react2.default.createElement(
                        _reactstrap.Col,
                        { key: index },
                        _react2.default.createElement(
                            'span',
                            {
                                onClick: function onClick() {
                                    return _this2.props.selectFrequency(type, cal.value);
                                },
                                className: freqClass },
                            cal.label
                        )
                    );
                });
            };
            return _react2.default.createElement(
                _reactstrap.Row,
                null,
                _react2.default.createElement(
                    _reactstrap.Col,
                    { md: 6, xs: 12 },
                    _react2.default.createElement(
                        _reactstrap.Row,
                        {
                            style: {
                                paddingBottom: "5px"
                            } },
                        _react2.default.createElement(
                            _reactstrap.Col,
                            null,
                            this.props.fields.weekdays.label
                        )
                    ),
                    _react2.default.createElement(
                        _reactstrap.Row,
                        { className: this.props.frequncyWidth },
                        getFrequency(this.props.allWeekDays, 1)
                    )
                ),
                _react2.default.createElement(
                    _reactstrap.Col,
                    { md: 6, xs: 12 },
                    _react2.default.createElement(
                        _reactstrap.Row,
                        {
                            style: {
                                paddingBottom: "5px"
                            } },
                        _react2.default.createElement(
                            _reactstrap.Col,
                            null,
                            this.props.fields.weeks.label
                        )
                    ),
                    _react2.default.createElement(
                        _reactstrap.Row,
                        { className: this.props.frequncyWidth },
                        getFrequency(this.props.allWeeks, 2)
                    )
                )
            );
        }
    }]);

    return FrequencyTab;
}(_react.Component);

exports.default = FrequencyTab;
