// @6d-ui/ui-helpers/build/components/routeVisits/VisitDetails.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactstrap = require('reactstrap');

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _SurveyForm = require('./SurveyForm');

var _SurveyForm2 = _interopRequireDefault(_SurveyForm);

var _fields = require('@6d-ui/fields');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Visitors = function (_Component) {
    _inherits(Visitors, _Component);

    function Visitors(props) {
        _classCallCheck(this, Visitors);

        var _this = _possibleConstructorReturn(this, (Visitors.__proto__ || Object.getPrototypeOf(Visitors)).call(this, props));

        var survay = _this.props.visitData ? _this.props.visitData.surveylist : null;
        var stock = _this.props.visitData ? _this.props.visitData.products : null;
        var survayData = [];

        if (survay) {
            survay.forEach(function (surv) {
                var sData = JSON.parse(surv.surveyData);
                if (sData.pages) {
                    sData.pages.forEach(function (page) {
                        page.elements.forEach(function (ele) {
                            survayData.push({ "title": ele.title, "answer": ele.answer });
                        });
                    });
                }
            });
        }

        _this.state = {
            activeTab: '1',
            survayData: survayData,
            stock: stock
        };
        _this.toggle = _this.toggle.bind(_this);
        return _this;
    }

    _createClass(Visitors, [{
        key: 'toggle',
        value: function toggle(tab) {
            this.state.activeTab !== tab && this.setState({ activeTab: tab });
        }
    }, {
        key: 'render',
        value: function render() {
            var _this2 = this;

            var getStockList = function getStockList() {
                if (_this2.state.stock) {
                    return _this2.state.stock.map(function (stock) {
                        return _react2.default.createElement(
                            _reactstrap.Row,
                            null,
                            _react2.default.createElement(
                                _reactstrap.Col,
                                { lg: '3' },
                                _react2.default.createElement(
                                    'div',
                                    { className: 'fs-16 p-4' },
                                    stock.productName
                                )
                            ),
                            _react2.default.createElement(
                                _reactstrap.Col,
                                { lg: '9' },
                                _react2.default.createElement(
                                    'div',
                                    { className: 'fs-16 fw-600 p-4' },
                                    stock.balance
                                )
                            )
                        );
                    });
                } else {
                    return _react2.default.createElement(
                        'div',
                        { className: 'text-center p-4' },
                        'Not Found !'
                    );
                }
            };
            if (this.props.visitData) {
                return _react2.default.createElement(
                    'div',
                    { className: 'p-3' },
                    _react2.default.createElement(
                        'div',
                        { className: 'd-flex flex-row mb-3' },
                        _react2.default.createElement('div', { style: { width: '70px', height: '70px', background: '#e1e1e1' }, className: 'profile-img' }),
                        _react2.default.createElement(
                            'div',
                            { className: 'py-1' },
                            _react2.default.createElement(
                                _reactstrap.Table,
                                { className: 'h-100', style: { lineHeight: 1 } },
                                _react2.default.createElement(
                                    'tbody',
                                    null,
                                    _react2.default.createElement(
                                        'tr',
                                        null,
                                        _react2.default.createElement(
                                            'td',
                                            { className: 'py-0 border-0 fs-12' },
                                            'Name'
                                        ),
                                        _react2.default.createElement(
                                            'th',
                                            { className: 'py-0 border-0 fs-14' },
                                            this.props.visitData.userName
                                        )
                                    ),
                                    _react2.default.createElement(
                                        'tr',
                                        null,
                                        _react2.default.createElement(
                                            'td',
                                            { className: 'py-0 border-0 align-middle fs-12' },
                                            'Designation'
                                        ),
                                        _react2.default.createElement(
                                            'th',
                                            { className: 'py-0 border-0 align-middle fs-14' },
                                            this.props.visitData.designation
                                        )
                                    ),
                                    _react2.default.createElement(
                                        'tr',
                                        null,
                                        _react2.default.createElement(
                                            'td',
                                            { className: 'py-0 border-0 align-bottom fs-12' },
                                            'Time Visited'
                                        ),
                                        _react2.default.createElement(
                                            'th',
                                            { className: 'py-0 border-0 align-bottom fs-14' },
                                            this.props.visitData.checkinDate
                                        )
                                    )
                                )
                            )
                        )
                    ),
                    _react2.default.createElement(
                        'div',
                        { className: 'form-tab' },
                        _react2.default.createElement(
                            _reactstrap.Nav,
                            { tabs: true },
                            _react2.default.createElement(
                                _reactstrap.NavItem,
                                null,
                                _react2.default.createElement(
                                    _reactstrap.NavLink,
                                    {
                                        className: (0, _classnames2.default)({ active: this.state.activeTab === '1' }),
                                        onClick: function onClick() {
                                            _this2.toggle('1');
                                        }
                                    },
                                    'Survey Form'
                                )
                            ),
                            _react2.default.createElement(
                                _reactstrap.NavItem,
                                null,
                                _react2.default.createElement(
                                    _reactstrap.NavLink,
                                    {
                                        className: (0, _classnames2.default)({ active: this.state.activeTab === '2' }),
                                        onClick: function onClick() {
                                            _this2.toggle('2');
                                        }
                                    },
                                    'Stock Info'
                                )
                            ),
                            _react2.default.createElement(
                                _reactstrap.NavItem,
                                null,
                                _react2.default.createElement(
                                    _reactstrap.NavLink,
                                    {
                                        className: (0, _classnames2.default)({ active: this.state.activeTab === '3' }),
                                        onClick: function onClick() {
                                            _this2.toggle('3');
                                        }
                                    },
                                    'Feedback'
                                )
                            )
                        ),
                        _react2.default.createElement(
                            _reactstrap.TabContent,
                            { activeTab: this.state.activeTab },
                            _react2.default.createElement(
                                _reactstrap.TabPane,
                                { tabId: '1' },
                                _react2.default.createElement(_SurveyForm2.default, { survayData: this.state.survayData })
                            ),
                            _react2.default.createElement(
                                _reactstrap.TabPane,
                                { tabId: '2' },
                                getStockList()
                            ),
                            _react2.default.createElement(
                                _reactstrap.TabPane,
                                { tabId: '3' },
                                _react2.default.createElement(
                                    _reactstrap.Container,
                                    { className: 'mt-3 bg-white border rounded' },
                                    _react2.default.createElement(
                                        _reactstrap.Row,
                                        { className: 'py-3' },
                                        _react2.default.createElement(_fields.FieldItem, {
                                            type: '11',
                                            width: 'xs',
                                            value: this.props.visitData.feedback
                                        })
                                    )
                                )
                            )
                        )
                    )
                );
            } else {
                return _react2.default.createElement(
                    'div',
                    { className: 'text-center p-4' },
                    'Not Found !'
                );
            }
        }
    }]);

    return Visitors;
}(_react.Component);

exports.default = Visitors;
